src/hostapi/asio/pa_asio.cpp

```diff
@@ -1880,22 +1893,35 @@ static PaError ValidateAsioSpecificStreamInfo(
 {
     if( streamInfo )
     {
-        if( streamInfo->size != sizeof( PaAsioStreamInfo )
-                || streamInfo->version != 1 )
+        switch( streamInfo->version )
         {
+        case 1:
+            /* NOTE: V1 structure's size is smaller by one pointer. */
+            if( streamInfo->size < sizeof( PaAsioStreamInfo ) - sizeof(PaAsio_MessageCallback*) )
```

Comment:
Shouldn't this do != instead of <?

Reply:
This particular check is difficult as we can't perfectly predict what the size of the original PaAsioStreamInfo structure would be without fully declaring it.  I use `<` instead of `!=` in case some padding is placed before the new field (although this is unlikely because the previous field is also a pointer).

Interested in input from maintainers on this issue.

Reply:
Also, you've increased the size of the struct by 2 pointers, not one.

Reply:
The internal struct is increased by two pointers (callbacks assigned to input and output stream).  I've only added one function pointer to PaAsioStreamInfo:  https://github.com/PortAudio/portaudio/pull/519/files#diff-d0136a1e498ae2f8b83825ed61b878a90f8904151f365099b498358d1937d863R218

Reply:
Nvm, I've looked at PaAsioStream not PaAsioStreamInfo.